R/install_binary_process.R

```diff
@@ -0,0 +1,32 @@
+
+make_install_process <- function(filename, lib = .libPaths()[[1L]],
+                                 metadata = NULL) {
+  filename; lib; metadata
```

Comment:
`filename` is not needed, but the other two are only used in a function that we create here and then keep around. In these cases what we almost always want is forcing an eval.

Often I just force all args if another function is created within the function and then passed around.

Reply:
It looks like `lib` is used in a call before creating `ppfun()` as well. But should the promise still be evaluated as expected regardless of when you force evaluation, `ppfun()` should have the body of `make_install_process()` as its enclosing environment right? When would relying on lazy evaluation of promises here fail?

I guess the error messages are probably potentially more clear if you force the promises earlier.

Reply:
True. But anyway, eager eval is always correct,  and lazy does not really give you anything here, so it is just easier to force all args, because the code can change with refactoring, and then it is easy to make a mistake about what has to be forced.

It fails if you make multiple processes with different metadata, because if we don't force, then metadata is only evaluated once the first process finishes,  and all processes started so far will get this metadata.